src/player.c

```diff
@@ -2167,11 +2173,89 @@ playback_next_bh(void *arg, int *retval)
   return COMMAND_END;
 }
 
+static int
+seek_calc_position_ms(struct player_seek_param *seek_param, struct db_queue_item **queue_item, int *position_ms)
+{
+  struct db_queue_item *seek_queue_item = NULL;
+  int seek_ms = 0;
+
+  // Initialize out parameters
+  *queue_item = NULL;
+  *position_ms = 0;
+
+  // Calculate seek position
+  if (seek_param->mode == PLAYER_SEEK_POSITION)
+    seek_ms = seek_param->ms;
+  else
+    seek_ms = pb_session.playing_now->pos_ms + seek_param->ms;
+
+  // Check if we need to switch to a previous track, this will be done if we are in the first 3 seconds
+  // of a track and we have a seek request for more than 3 seconds
+  if (seek_ms < 0)
+    {
+      if (pb_session.playing_now->pos_ms < 3000)
+	{
+	  // We are in the first 3 seconds of the track, switch to the previous track and recalculate the absolute seek position
+	  seek_queue_item = queue_item_prev(pb_session.playing_now->item_id);
+
+	  if (seek_queue_item)
+	    {
+	      seek_ms = seek_queue_item->song_length + seek_ms;
+	      // Make sure to not try to seek behind the previous track (this is also the case if song_length is zero)
+	      seek_ms = (seek_ms < 0) ? 0 : seek_ms;
+	    }
+	  else
+	    {
+	      // There is no previous queue item, seek to the start of the current item
+	      seek_ms = 0;
+	    }
+	}
+      else
+	{
+	  // We are more than 3 seconds into the playing track, seek to beginning of current track
+	  seek_ms = 0;
+	}
+    }
+  else if (seek_ms > 0 && seek_ms > pb_session.playing_now->len_ms)
+    {
+      // We are seeking beyond the current track, play the next track from the beginning
+      seek_queue_item = queue_item_next(pb_session.playing_now->item_id);
+      if (seek_queue_item)
+        {
+	  seek_ms = 0;
+	}
+    }
+
+  if (!seek_queue_item)
+    {
+      // Seeking in the current queue item
+      seek_queue_item = db_queue_fetch_byitemid(pb_session.playing_now->item_id);
+    }
+
+  if (!seek_queue_item)
+    {
+      DPRINTF(E_LOG, L_PLAYER, "Error fetching queue item for seek command (seek_ms=%d, seek_mode=%d)\n", seek_param->ms, seek_param->mode);
+      return -1;
+    }
+
+  if (seek_ms < 0)
```

Comment:
Is this a possible outcome?

Reply:
It is, if there is a error in the code above :-)
Thought, i play it safe here and have a last sanity check for the seek position.

Reply:
Ok, I see. I would suggest removing it, because if is no path to it then it is dead code that might also just get optimized away by the compiler.

Reply:
done